frontend/packages/core/src/NPS/feedback.tsx

```diff
@@ -0,0 +1,219 @@
+import { client } from "../Network";
+import type { ClutchError } from "../Network/errors";
+import { Typography } from "../typography";
+
+import EmojiRatings, { Rating } from "./emojiRatings";
+
+/** Interfaces */
+
+interface FeedbackOptions {
+  origin: "ORIGIN_UNSPECIFIED" | "WIZARD" | "ANYTIME";
+  onSubmit?: (submit: boolean) => void;
+}
+
+// Defaults in case of API failure
+export const defaults: IClutch.feedback.v1.ISurvey = {
+  prompt: "Rate Your Experience",
+  freeformPrompt: "What would you recommend to improve this?",
+  ratingLabels: [
+    {
+      emoji: 1,
+      label: "bad",
+    },
+    {
+      emoji: 2,
+      label: "ok",
+    },
+    {
+      emoji: 3,
+      label: "great",
+    },
+  ],
+};
+
+const StyledButton = styled(Button)({
+  fontSize: "14px",
+  padding: "0 8px",
+  height: "18px",
+});
+
+const StyledTextField = styled(TextField)({
+  marginTop: "15px",
+  ".MuiInputBase-root": {
+    fontSize: "14px",
+  },
+});
+
+/**
+ * NPS feedback component which is the base for both Wizard and Anytime
+ * Will fetch given survey options from the server based on the given origin
+ * Then display a feedback component based on the given emoji ratings
+ *
+ * @param opts Available feedback options
+ * @returns NPSFeedback component
+ */
+const NPSFeedback = (opts: FeedbackOptions = { origin: "ORIGIN_UNSPECIFIED" }) => {
+  const [hasSubmit, setSubmit] = useState<boolean>(false);
+  const [selected, setSelected] = useState<Rating>(null);
+  const [feedback, setFeedback] = useState<string>("");
+  const [error, setError] = useState<boolean>(false);
+  const [survey, setSurvey] = useState<IClutch.feedback.v1.ISurvey>({});
+  const [requestId, setRequestId] = useState<string>("");
+  const maxLength = 180;
+  const debounceTimer = 500;
+
+  const trimmed =
+    feedback.trim().length > maxLength
+      ? `${feedback.trim().substring(0, maxLength - 3)}...`
+      : feedback;
+
+  /** Property objects used to extend components and remove console warnings */
+  const AlertProps = {
+    iconMapping: {
+      info: <MuiSuccessIcon style={{ color: "#3548d4" }} />,
+    },
+    style: {
+      margin: "32px",
+      alignItems: "center",
+    },
+  };
+
+  const textFieldProps = {
+    fullWidth: true,
+    InputProps: {
+      rows: 3,
+      rowsMax: 3,
+    },
+  };
+
+  // Will fetch the survey results for the given origin on load
+  React.useEffect(() => {
+    let data: IClutch.feedback.v1.ISurvey = defaults;
+
+    client
+      .post("/v1/feedback/getSurveys", {
+        origins: [opts.origin],
+      })
+      .then(response => {
+        const surveyData: IClutch.feedback.v1.IGetSurveysResponse = response?.data?.originSurvey;
+
+        data = surveyData[opts.origin] ?? defaults;
+      })
+      .catch((err: ClutchError) => {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      })
+      .finally(() => {
+        setRequestId(uuid());
+        setSurvey(data);
```

Comment:
nice! didn't know about the `finally` method

Reply:
Yep! It's extremely useful in promise chains